genesyscloud/group/genesyscloud_group_proxy.go

```diff
@@ -0,0 +1,185 @@
+	return p.groupsApi.PostGroups(*group)
+}
+
+func updateGroupFn(ctx context.Context, p *groupProxy, id string, group *platformclientv2.Groupupdate) (*platformclientv2.Group, *platformclientv2.APIResponse, error) {
+	return p.groupsApi.PutGroup(id, *group)
+}
+
+func deleteGroupFn(ctx context.Context, p *groupProxy, id string) (*platformclientv2.APIResponse, error) {
+	return p.groupsApi.DeleteGroup(id)
+}
+
+func getGroupByIdFn(ctx context.Context, p *groupProxy, id string) (*platformclientv2.Group, *platformclientv2.APIResponse, error) {
+	group := rc.GetCache(p.groupCache, id)
+	if group != nil {
+		return group, nil, nil
+	}
+	return p.groupsApi.GetGroup(id)
+}
+func addGroupMembersFn(ctx context.Context, p *groupProxy, id string, members *platformclientv2.Groupmembersupdate) (*interface{}, *platformclientv2.APIResponse, error) {
+	return p.groupsApi.PostGroupMembers(id, *members)
+}
+
+func deleteGroupMembersFn(ctx context.Context, p *groupProxy, id string, members string) (*interface{}, *platformclientv2.APIResponse, error) {
+	return p.groupsApi.DeleteGroupMembers(id, members)
+}
+
+func getGroupMembersFn(ctx context.Context, p *groupProxy, id string) (*[]string, *platformclientv2.APIResponse, error) {
+	members, response, err := p.groupsApi.GetGroupIndividuals(id)
+
+	if err != nil {
+		return nil, response, err
+	}
+
+	var existingMembers []string
+	if members.Entities != nil {
+		for _, member := range *members.Entities {
+			existingMembers = append(existingMembers, *member.Id)
+		}
+	}
+	return &existingMembers, nil, nil
+}
+
+func getGroupByNameFn(ctx context.Context, p *groupProxy, name string) (*platformclientv2.Groupssearchresponse, *platformclientv2.APIResponse, error) {
+	exactSearchType := "EXACT"
+	nameField := "name"
+	nameStr := name
+
+	searchCriteria := platformclientv2.Groupsearchcriteria{
+		VarType: &exactSearchType,
+		Value:   &nameStr,
+		Fields:  &[]string{nameField},
+	}
+
+	groups, resp, getErr := p.groupsApi.PostGroupsSearch(platformclientv2.Groupsearchrequest{
+		Query: &[]platformclientv2.Groupsearchcriteria{searchCriteria},
+	})
+
+	return groups, resp, getErr
+}
+
+func getAllGroupFn(ctx context.Context, p *groupProxy) (*[]platformclientv2.Group, *platformclientv2.APIResponse, error) {
+	var allGroups []platformclientv2.Group
+	const pageSize = 100
+
+	groups, resp, getErr := p.groupsApi.GetGroups(pageSize, 1, nil, nil, "")
+	if getErr != nil {
+		return nil, resp, fmt.Errorf("Failed to get first page of groups: %v", getErr)
+	}
+
+	allGroups = append(allGroups, *groups.Entities...)
+
+	for pageNum := 2; pageNum <= *groups.PageCount; pageNum++ {
+		groups, resp, getErr := p.groupsApi.GetGroups(pageSize, pageNum, nil, nil, "")
+		if getErr != nil {
+			return nil, resp, fmt.Errorf("Failed to get page of groups: %v", getErr)
+		}
+		allGroups = append(allGroups, *groups.Entities...)
+	}
+
+	for _, group := range allGroups {
```

Comment:
@HemanthDogiparthi12 Added caching to this code.  Nice job on the caching, very clean